core/net/net.go

```diff
@@ -48,6 +48,9 @@ type API interface {
 	// DeleteThread with credentials.
 	DeleteThread(ctx context.Context, creds thread.Credentials) error
 
+	// GetThreadAddresses returns the host addresses augmented with peerID and threadID components
+	GetThreadAddresses(ctx context.Context, creds thread.Credentials) ([]ma.Multiaddr, error)
```

Comment:
Added this new method to the `Net` that returns addresses for a specified thread.